ci/generate_docs.sc

```diff
@@ -246,11 +260,29 @@ def copyMarathon(buildDir: Path): Path = {
   buildDir / 'marathon
 }
 
-def buildDocs(buildDir: Path, docsDir: Path, checkedRepoDir: Path) = {
+def buildDocs(buildDir: Path, docsDir: Path, checkedRepoDir: Path): Path = {
   prepareDockerImage(buildDir, docsDir)
   val versionedDocsDirs = checkoutDocsToTempFolder(buildDir, docsDir, checkedRepoDir)
   generateTopLevelDocs(buildDir, docsDir, checkedRepoDir)
   generateVersionedDocs(buildDir, versionedDocsDirs)
+  val siteDir = (buildDir/'docs/'_site)
+  siteDir
+}
+
+def publishToGithub(siteDir: Path, buildDir: Path, remote: String): Unit = {
+  println("Publishing docs to github")
+  %('git, 'clone, "-b", "gh-pages", "--single-branch", remote, "marathon-gh-pages")(buildDir)
+  val ghPagesDir = buildDir/"marathon-gh-pages"
+  // delete the old docs but not .git files
+  ls! ghPagesDir |? (path => !path.last.startsWith(".")) | (path => rm! path)
```

Comment:
Doesn't this clean all folders? Do we really want this? I thought we would release on version not all.

Reply:
Yes, it cleans everything. I believe that it's more reliable to republish the documentation every time, simply because converging the state might get more complicated when we use `push the diff` approach. For example there might be layout changes which might not be applied to some versions if we publish them separately.